qiskit_aer/quantum_info/states/aer_statevector.py

```diff
@@ -113,7 +113,12 @@ def sample_memory(self, shots, qargs=None):
         else:
             qubits = np.array(qargs)
         self._aer_state.close()
-        self._aer_state = AerState(**self._aer_state.configuration())
+
+        configs = self._aer_state.configuration()
+        if 'seed_simulator' in configs:
+            configs['seed_simulator'] = int(configs['seed_simulator']) + 1
+        self._aer_state = AerState(**configs)
```

Comment:
This seems like a slightly odd split in responsibilities.  Perhaps it would be more convenient for now, and for future development if `AerState` gains an `advance_rng` method or something like that mutates the `AerState` in place.  It'll be easier to re-use between the different Aer `quantum_info` classes you're planning that way, it gives clearer hints as to what's happening, and modifying it is kind of an `AerState` internal feature that we don't particularly want client classes to need to know about.

Reply:
`QuantumState` advances RNG. So, `AerStatevector` also needs to advance it in `sample_memory()`.
https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/quantum_info/states/quantum_state.py#L269

However, instead of manually update of `seed_simulator` by increasing `1`, it is better to succeed the state of `mt19937` from the previous `AerState`.